build/cloudbuild-tf-apply.yaml

```diff
@@ -22,8 +22,11 @@ steps:
   args:
   - -c
   - |
-    echo "Setting up gcloud for impersonation"
-    gcloud config set auth/impersonate_service_account ${_TF_SA_EMAIL}
+   tf_sa_email=${_TF_SA_EMAIL}
```

Comment:
we could probably just use the substitution directly, rather than setting to a variable first

Reply:
I was initially trying to use it directly like `if [[ -n ${_TF_SA_EMAIL} ]]; then` , but if `_TF_SA_EMAIL` is not set (in the case of app infra pipelines) cloudbuild fails to start build and complains about  a missing substitution. I tried this as a workaround which seemed to work.